packages/vue-client/src/components/Timer.vue

```diff
@@ -0,0 +1,39 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+
+const props = defineProps<{
+    countsDown: boolean;
+    remainingTimeMS: number;
+}>();
+
+let t = ref(props.remainingTimeMS);
+
+if (props.countsDown) {
+    const timerIndex = setInterval(() => {
+        if (t.value <= 0) {
+            clearInterval(timerIndex);
+        }
+        else {
+        t.value -= 1000;
+        }
+    }, 1000)
+}
+
+function msToTime(ms: number): string {
+  //ToDo: improve, this is kinda horrible
+  let msCopy = ms;
+  let days = Math.floor((msCopy / (1000 * 60 * 60 * 24)))
+  msCopy = msCopy - days * (1000 * 60 * 60 * 24);
+  let hours = Math.floor((msCopy / (1000 * 60 * 60)))
+  msCopy = msCopy - hours * (1000 * 60 * 60);
+  let minutes = Math.floor(msCopy / (1000 * 60));
+  msCopy = msCopy - minutes * (1000 * 60);
+  let seconds = Math.floor(msCopy / 1000);
```

Comment:
Maybe it would be less verbose to go the other way round? Calculate seconds first?

Reply:
right, I changed it accordingly. I was mainly thinking there is probably a "proper" way of doing this, like a method of Date or so, that would create a formatted string. But for now I'm actually happy with what I got now.